### ISSUE TITLE
Results on Cityscapes

### ISSUE THREAD

**Issue by zyy-cn** [janghyuncho/PiCIE#15]
Hi Cho:

   It's me again:) This time I try to reproduce the result on Cityscapes, which are 65.50 acc and 12.31 mIoU that reported in your paper, by doing the following steps:

1. Download Cityscapes data 'gtFine_trainvaltest.zip' and 'leftImg8bit_trainvaltest.zip' from the official Cityscapes website.

2. Make a projection that change the indices in each 'xxx_gtFine_labelIds.png' by the following dictionary that maps indices from [-1, 33] to [-1, 26]:
dict = {0: -1, 1: -1, 2: -1, 3: -1, 4: -1, 5: -1, 6: -1, 7: 0, 8: 1, 9: 2, 10: 3, 11: 4, 12: 5, 13: 6, 14: 7, 15: 8, 16: 9, 17: 10, 18: 11, 19: 12, 20: 13, 21: 14, 22: 15, 23: 16, 24: 17, 25: 18, 26: 19, 27: 20, 28: 21, 29: 22, 30: 23, 31: 24, 32: 25, 33: 26, -1: -1}

3. Keep all the rest hyperparameters the same as what they are in 'train_picie.py' and 'train_picie.sh'.

4. Use 2975 training samples for training, and 500 validation samples for evaluation.

However, I cannot get the desired results but only got 7.293 +/- 0.074 and 34.779 +/- 0.350 for mIoU and accuracy, respectively. 

I wonder if I took the right way to reproduce the results on Cityscapes, or some additional modifications to the hyperparameters or code are needed to get the right results?

Thanks!

**Comment by vobecant** [comment]
Hi @zyy-cn , I am also interested in replicating and evaluating the results of PiCIE on Cityscapes. Can you please tell me how exactly do you evaluate? Do you do the Hungarian matching? Can you please provide me the trained network? Thank you very much in advance.

**Comment by zyy-cn** [comment]
Hi @vobecant 

The question you asked is what confused me. The range of indices in 'gtFine_trainvaltest.zip' is from -1 to 33, **35 classes in total, not 30 classes as mentioned in the paper**, so I'm not sure if I got the right way to reproduce this work. Specifically, I follow the definition [here](https://github.com/mcordts/cityscapesScripts/blob/master/cityscapesscripts/helpers/labels.py) to filter out the 'void' group and index '-1', resulting in 27 classes for evaluation. I use the train/evaluate code provided on this repo and replace the dictionary in 'fine_to_coarse_dict.pickle' to the one I listed above, the indices in prediction are reordered to match the ground-truth with Hungarian matching, as @janghyuncho did in his code.

**Comment by janghyuncho** [comment]
Hi zyy-cn, 

Thank you for your interest in PiCIE! And I am sorry for the delayed reply; I have been very busy lately with deadline. 

The training setup for Cityscapes has a bit of difference from COCO, but more importantly we train the model on "train_extra" and "test" subsets which is far more than "train" subset of the dataset. I will upload training and testing code in a few weeks after I get less busy. I appreciate your patience. 

In the meantime, [here](https://urldefense.com/v3/__https://drive.google.com/file/d/1phk0hlpFT2RMmOqzQ6Csoj6HgnZFskbK/view?usp=sharing__;!!OToaGQ!4f5kMMLjSy4HWv9ov9Th7sF3SGJ4ZfvKkeQjRLpqnPPoZxGh7Hu5Lcqhg6F-awOuR4lG$) is the pre-trained weight. 

Thank you!

**Comment by janghyuncho** [comment]
Hi, I uploaded the training end evaluation code for cityscapes for [picie](https://github.com/janghyuncho/PiCIE/blob/master/train_picie.py) and [mdc](https://github.com/janghyuncho/PiCIE/blob/master/train_mdc.py) ([log](https://github.com/janghyuncho/PiCIE/tree/master/logs/reproduce)). 

Sorry for the delay. 

Thank you!
